### PR TITLE
Fixes #16406 - Adds 'builtin' attribute to Role list

### PR DESCRIPTION
**Pull request by shlomizadok**
@tstrachota  Haven't found tests for role. Please advise

### CODE REVIEW

**Comment by ares** [comment]
Somehow I'm getting the output twice

```sh
ares@tony:[~/z/hammer-cli][master✓]$ be bin/hammer role info --id 27
Id:      27
Name:    a
Builtin: 
Builtin: No

ares@tony:[~/z/hammer-cli][master✓]$ be bin/hammer role info --id 11
Id:      11
Name:    Viewer
Builtin: 0
Builtin: No
```

**Comment by ares** [comment]
also I think it would be nicer if listing would print `Yes` and `No` values instead of numbers

```
ares@tony:[~/z/hammer-cli][master✓]$ be bin/hammer role list        
---|--------------------------|--------
ID | NAME                     | BUILTIN
---|--------------------------|--------
27 | a                        |        
62 | asdasd                   | 0      
17 | Compliance manager       | 0      
16 | Compliance viewer        | 0      
18 | Create ARF report        | 0      
13 | Default role             | 2      
21 | Discovery                | 0      
6  | Discovery Manager        | 0      
5  | Discovery Reader         | 0      
10 | Edit hosts               | 0      
8  | Edit partition tables    | 0      
31 | host_exposer             | 0      
7  | Manager                  | 0      
57 | Manager of Aress         | 0      
19 | org_delegator            | 0      
20 | param_test               | 0      
4  | Remote Execution Manager | 0      
3  | Remote Execution User    | 0      
12 | Site manager             | 0      
1  | Tasks Manager            | 0      
2  | Tasks Reader             | 0      
11 | Viewer                   | 0      
9  | View hosts               | 0      
---|--------------------------|-------
```

**Comment by shlomizadok** [comment]
@ares - Thanks for testing this. 
I have DRYed the code, changed to `Yes/No` and handled the duplication

**Comment by ares** [comment]
Works fine for me, I don't have permission on this repo, so hopefully some @theforeman/hammer-admin can take a look too and merge.

File: lib/hammer_cli_foreman/role.rb [hunk 1]
```diff
@@ -6,25 +6,28 @@ class Role < HammerCLIForeman::Command
...
+      end
+    end
+
     class ListCommand < HammerCLIForeman::ListCommand
+      include RoleDataExtender
+
       output do
         field :id, _("Id")
         field :name, _("Name")
+        field :builtin_text, _("Builtin")
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/role.rb]: I fixed boolean formatter in hammer-cli: https://github.com/theforeman/hammer-cli/pull/219
Would you mind using
```ruby
field :builtin, _("Builtin"), Fields::Boolean
```
instead to keep commands consistent? It should work fine with the fix above.

File: lib/hammer_cli_foreman/role.rb [hunk 2]
```diff
@@ -6,25 +6,28 @@ class Role < HammerCLIForeman::Command
 
     resource :roles
 
+    module RoleDataExtender
+      def extend_data(data)
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/role.rb]: Extending data shouldn't be necessary now, see my comment below, please.

**Comment by shlomizadok** [comment]
@tstrachota, changed. Thanks !

File: lib/hammer_cli_foreman/role.rb [hunk 3]
```diff
@@ -10,21 +10,14 @@ class ListCommand < HammerCLIForeman::ListCommand
       output do
         field :id, _("Id")
         field :name, _("Name")
+        field :builtin_text, _("Builtin"),  Fields::Boolean
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/role.rb]: It should be only
```ruby
field :builtin, _("Builtin"),  Fields::Boolean
```
`:builtin_text` ho longer exists now :)

Comment (shlomizadok) [inline lib/hammer_cli_foreman/role.rb]: ooops, changed

**Comment by tstrachota** [comment]
Merged, thanks @shlomizadok !

**Pull request merged**